spec/new_integration/rack_app/rack_app_spec.rb

```diff
@@ -100,10 +122,243 @@ def handle(*, res)
+            module Home
+              class Index < Hanami::Action
+                def handle(*, res)
+                  res.body = "Hello from Slice"
+                end
+              end
+            end
+          end
+        end
+      RUBY
+
+      require "hanami/boot"
+
+      get "/"
+
+      expect(last_response.status).to eq 200
+      expect(last_response.body).to eq "Hello from App"
+    end
+  end
+
+  xspecify "It does not choose actions from slice to route in app" do
+    with_tmp_directory(Dir.mktmpdir) do
+      write "config/application.rb", <<~RUBY
+        require "hanami"
+
+        module TestApp
+          class Application < Hanami::Application
+            config.logger.stream = File.new("/dev/null", "w")
+          end
+        end
+      RUBY
+
+      write "config/routes.rb", <<~RUBY
+        module TestApp
+          class Routes < Hanami::Routes
+            define do
+              get "/feedbacks", to: "feedbacks.index"
+
+              slice :api, at: "/api" do
+                get "/people", to: "people.index"
+              end
+            end
+          end
+        end
+      RUBY
+
+      write "app/actions/feedbacks/index.rb", <<~RUBY
+        require "hanami/action"
+
+        module TestApp
+          module Actions
+            module Feedbacks
+              class Index < Hanami::Action
+                def handle(*, res)
+                  res.body = "Feedbacks"
+                end
+              end
+            end
+          end
+        end
+      RUBY
+
+      write "slices/api/actions/people/index.rb", <<~RUBY
+        require "hanami/action"
+
+        module Api
+          module Actions
+            module People
+              class Index < Hanami::Action
+                def handle(*, res)
+                  res.body = "People"
+                end
+              end
+            end
+          end
+        end
+      RUBY
+
+      require "hanami/boot"
+
+      get "/api/people"
+
+      expect(last_response.status).to eq 200
+      expect(last_response.body).to eq "People"
+    end
+  end
+
+  specify "It doesn't boot the app, if referenced app action isn't registered" do
+    with_tmp_directory(Dir.mktmpdir) do
+      write "config/application.rb", <<~RUBY
+        require "hanami"
+
+        module TestApp
+          class Application < Hanami::Application
+          end
+        end
+      RUBY
+
+      write "config/routes.rb", <<~RUBY
+        module TestApp
+          class Routes < Hanami::Routes
+            define do
+              get "/missing", to: "missing.action"
+            end
+          end
+        end
+      RUBY
+
+      expect { require "hanami/boot" }.to raise_error do |exception|
+        expect(exception).to be_kind_of(Hanami::Application::Routing::UnknownActionError)
+        expect(exception.message).to include("missing.action")
+      end
+    end
+  end
+
+  xspecify "It doesn't boot the app, if referenced slice action isn't registered" do
```

Comment:
@timriley This is connected to https://github.com/hanami/hanami/pull/1177/commits/350e03ab7dc20b8406a9ec67e287dcbd53267a6a

Reply:
Addressed this one (along with a range of related issues) in https://github.com/hanami/hanami/pull/1177/commits/9821647f97df6386f7e0b7e9bd8f485510645dd8